populator-machinery/controller_test.go

```diff
@@ -715,14 +811,47 @@ func TestSyncPvcWithProviderImplementation(t *testing.T) {
 			pvcName:      testPvcName,
 			initialObjects: []runtime.Object{
 				pvc(testPvcName, testPvcNamespace, "", testStorageClassName, "",
-					dsf(testApiGroup, testDatasourceKind, testDataSourceName, testPvcNamespace), ""),
+					dsf(testApiGroup, testDatasourceKind, testDataSourceName, testPvcNamespace), "", v1.ReadWriteOnce),
 				ust(),
 				sc(testStorageClassName, storagev1.VolumeBindingImmediate),
-				pvc(testPvcPrimeName, testVpWorkingNamespace, "", testStorageClassName, "", nil, corev1.ClaimPending),
 			},
-			populateFn:     populateOperationStartError,
-			expectedResult: nil,
-			expectedKeys:   []string{pvcPrimeKey},
+			populateFn:            populateOperationStartError,
+			expectedResult:        nil,
+			expectedKeys:          []string{pvcPrimeKey},
+			expectedMutateObjects: mutateObjects{pvcPrime: pvc(testPvcPrimeName, testVpWorkingNamespace, "", testStorageClassName, testPvName, nil, "", v1.ReadWriteOnce)},
+		},
+		{
+			name:         "Create PVC prime mutate successd",
+			key:          "pvc/" + testPvcNamespace + "/" + testPvcName,
+			pvcNamespace: testPvcNamespace,
+			pvcName:      testPvcName,
+			initialObjects: []runtime.Object{
+				pvc(testPvcName, testPvcNamespace, "", testStorageClassName, "",
+					dsf(testApiGroup, testDatasourceKind, testDataSourceName, testPvcNamespace), "", v1.ReadWriteOnce),
+				ust(),
+				sc(testStorageClassName, storagev1.VolumeBindingImmediate),
+			},
+			pvcPrimeMutator:       pvcPrimeMutateSuccess,
```

Comment:
Can you make the mutator change the pvc in a more significant way? Otherwise it's hard to tell if the mutator is actually being called (eg: I think you could set pvcPrimeMutator to nil, and this test would still pass)